src/hotspot/share/gc/z/zOopClosures.hpp

```diff
@@ -67,14 +67,22 @@ class ZPhantomIsAliveObjectClosure : public BoolObjectClosure {
 
 class ZPhantomKeepAliveOopClosure : public ZRootsIteratorClosure {
 public:
-  virtual void do_oop(oop* p);
-  virtual void do_oop(narrowOop* p);
+  virtual void do_oop(oop* p) override;
+  virtual void do_oop(narrowOop* p) override;
+
+  virtual ZNMethodEntry nmethod_entry() const override {
+    return ZNMethodEntry::VerifyDisarmed;
```

Comment:
I can't see that this will trigger any verification. The ZPhantomKeepAliveOopClosure is used in ZProcessWeakRootsTask, and it does not visit nmethods. Even if it did, it could still not assert that the visited nmethods are not disarmed, because it could have a single oop that is live, despite the nmethod not being entered.